app/views/rewards/index.html.erb

```diff
@@ -19,6 +19,7 @@
           <td><%= reward.description %></td>
           <td><%= number_to_currency(reward.price) %></td>
           <td><%= link_to 'Show', reward %></td>
+          <td><%= link_to 'Buy', new_order_path(reward: reward) %></td>
```

Comment:
consider simply sending a create request - instead of rendering the form, in which you can really change anything, only to click button to create order :)